packages/commonwealth/client/scripts/views/components/ProposalSelector/ProposalSelectorItem.tsx

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import { CWCheck } from 'views/components/component_kit/cw_icons/cw_icons';
+import { chainEntityTypeToProposalName } from 'identifiers';
+import { CWText } from '../component_kit/cw_text';
+import { IAaveProposalResponse } from 'adapters/chain/aave/types';
+import { ICompoundProposalResponse } from 'adapters/chain/compound/types';
+import { IChainEntityKind } from 'chain-events/src';
+
+interface ProposalSelectorItemProps {
+  proposal: IAaveProposalResponse | ICompoundProposalResponse;
+  isSelected: boolean;
+  onClick: (
+    proposal: IAaveProposalResponse | ICompoundProposalResponse
+  ) => void;
+}
+
+const ProposalSelectorItem = ({
+  onClick,
+  proposal,
+  isSelected,
+}: ProposalSelectorItemProps) => {
+  return (
+    <div className="chain-entity" onClick={() => onClick(proposal)}>
+      <div className="selected">{isSelected && <CWCheck />}</div>
+      <div className="text">
+        <CWText fontWeight="medium" truncate noWrap>
+          {chainEntityTypeToProposalName('proposal' as IChainEntityKind) +
+            (proposal.identifier.startsWith('0x')
+              ? ` ${proposal.identifier.slice(0, 6)}...`
```

Comment:
What is the scenario where a proposal identifier is a hex value? If there is a hex scenario is it representative of an int proposal id that we can decode to remain human readable?

Reply:
OpenZeppelin governance uses a hash of proposal data as the proposal ID so I don't think there is a human readable version. OZ hashing docs: https://docs.openzeppelin.com/contracts/4.x/api/governance#Governor-hashProposal-address---uint256---bytes---bytes32-

Reply:
That makes sense, just looking here I know that downstream from proposalId creation its passed as a uint so just assumed it was used as an int at some point: 
https://docs.openzeppelin.com/contracts/4.x/api/governance#Governor-state-uint256-

Reply:
Actually in the link you sent the proposal hash is also returned as uint256. At what point are we getting a bytes32 style hash for proposalId?

Reply:
Hm yeah, I think at some point we convert to hex because it's just a realllly long number and is even less readable than hex but will double check.

Reply:
Yeah that makes sense then, a hash is good if this is the case. This shouldnt need any changes